api/tests/Feature/NotificationTest.php

```diff
@@ -0,0 +1,182 @@
+use Database\Seeders\RolePermissionSeeder;
+use Database\Seeders\SkillFamilySeeder;
+use Database\Seeders\SkillSeeder;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Nuwave\Lighthouse\Testing\MakesGraphQLRequests;
+use Nuwave\Lighthouse\Testing\RefreshesSchemaCache;
+use Tests\TestCase;
+
+class NotificationTest extends TestCase
+{
+    use MakesGraphQLRequests;
+    use RefreshDatabase;
+    use RefreshesSchemaCache;
+
+    protected $user;
+
+    protected $notification;
+
+    protected $originalStatus;
+
+    protected $newStatus;
+
+    protected $pool;
+
+    protected $queryNotifications = /** GraphQL */ '
+        query Notifications {
+            notifications {
+                data {
+                    id
+                    readAt
+                    ... on PoolCandidateStatusChangedNotification {
+                        oldStatus
+                        newStatus
+                    }
+                }
+            }
+        }
+    ';
+
+    protected function setUp(): void
+    {
+
+        parent::setUp();
+
+        $this->seed(ClassificationSeeder::class);
+        $this->seed(GenericJobTitleSeeder::class);
+        $this->seed(SkillFamilySeeder::class);
+        $this->seed(SkillSeeder::class);
+        $this->seed(RolePermissionSeeder::class);
+
+        $this->user = User::factory()
+            ->asApplicant()
+            ->create();
+
+        $this->pool = Pool::factory()
+            ->published()
+            ->create();
+
+        $this->originalStatus = PoolCandidateStatus::NEW_APPLICATION->name;
+        $this->newStatus = PoolCandidateStatus::PLACED_TERM->name;
+        $this->user->notify(
+            new PoolCandidateStatusChanged(
+                $this->originalStatus,
+                $this->newStatus,
+                $this->pool->id,
+                $this->pool->name
+            )
+        );
+
+        $this->notification = $this->user->unreadNotifications()->first();
+    }
+
+    public function testQueryNotification(): void
+    {
+
+        $this->actingAs($this->user, 'api')
+            ->graphQL($this->queryNotifications)->assertJson([
+                'data' => [
+                    'notifications' => [
+                        'data' => [
+                            [
+                                'id' => $this->notification->id,
+                                'readAt' => null,
+                            ],
+                        ],
+                    ],
+                ],
+            ]);
+    }
+
+    public function testUnreadNotification(): void
+    {
+        $response = $this->actingAs($this->user, 'api')
+            ->graphQL(/** @lang GraphQL */ '
+                mutation unReadNotification($id: UUID!) {
+                    markNotificationAsUnread(id: $id) {
+                        id
+                        readAt
+                    }
+                }
+            ', ['id' => $this->notification->id]);
```

Comment:
Doesn't this notification start out as unread? I'd set read_at to non-null before running the mutation.

Reply:
lol it was even worse than that... I was attempting to grab the wrong field out of the response so it was always going to be null 😅 

Fixed in [add read test, fix unread test](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9694/commits/4f77c45161b860733739143bc899a8a1d7c9da34)

Reply:
Okay, that's good, but this notification still _starts out_ as unread, doesn't it? 😆

Reply:
It does but as it stands, our tests are ran sequentially (default unless you turn on parallel tests). So the previous test that marks it as read is ran first and then we mark it as unread here. So it goes like this where each case is testing one thing in sequential order.

1. Query notification
2. Mark as read
3. Mark as unread
4. Delete

I did this so we don't have to keep massaging data or create new data for each test case and test at the same time as creating the data for the next test. 

Though, we could assume they are not ran sequentially and just massage the data in each case or merge them into a single test that does read/unread.

Reply:
Hmmm. Are you sure about that? I was under the impression that by using the `use RefreshDatabase;` trait, the database is reset and the `setUp` function is run before every single test function. :/

Reply:
I have seriously been writing PHPUnit tests based off of a false assumption for years and I'm honestly shocked it hasn't bitten me yet 😆 I always thought

- Class = Test
- Method = Test case

But, it is the opposite! So when laravel told me that it refreshes on every test, I assumed after all methods in the class were called 😬 

I have updated the test to mark the notification as read at the start 😅 [ensure notifications is read before testing unread mutation](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9694/commits/14dd2688874139c07295849e501910c34b07b591)